src/robot/running/context.py

```diff
@@ -67,6 +76,7 @@ def __init__(self, suite, namespace, output, dry_run=False):
         self.timeout_occurred = False
         self.steps = []
         self.user_keywords = []
+        self._event_loop = event_loop
```

Comment:
This attribute needs to be accessed by `ExecutionContexts` so it shouldn't have `_` prefix. I believe it could be just `self.event_loop` and then there could be a method like `get_running_event_loop()` (or a property like `running_event_loop`) that creates a new loop and returns that.